File: app/src/main/java/com/enipro/data/remote/model/Education.java
package com.enipro.data.remote.model;


import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import org.parceler.ParcelConstructor;

@org.parceler.Parcel
public class Education {

    @SerializedName("_id")
    @Expose
    public ObjectId _id;

    @SerializedName("school")
    @Expose
    public String school;

    @SerializedName("course")
    @Expose
    public String course;

    @SerializedName("degree")
    @Expose
    public String degree;

    @SerializedName("from")
    @Expose
    public String from;

    @SerializedName("to")
    @Expose
    public String to;


    @ParcelConstructor
    public Education(String school, String course, String degree, String from, String to) {
        this.school = school;
        this.course = course;
        this.degree = degree;
        this.from = from;
        this.to = to;
    }

    public ObjectId get_id() {
        return _id;
    }

    public void set_id(ObjectId _id) {
        this._id = _id;
    }

    public String getSchool() {
        return school;
    }

    public void setSchool(String school) {
        this.school = school;
    }

    public String getCourse() {
        return course;
    }

    public void setCourse(String course) {
        this.course = course;
    }

    public String getDegree() {
        return degree;
    }

    public void setDegree(String degree) {
        this.degree = degree;
    }

    public String getFrom() {
        return from;
    }

    public void setFrom(String from) {
        this.from = from;
    }

    public String getTo() {
        return to;
    }

    public void setTo(String to) {
        this.to = to;
    }
}

File: app/src/main/java/com/enipro/data/remote/model/Education.kt
package com.enipro.data.remote.model


import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Education
constructor(@field:SerializedName("school")
            @field:Expose
            var school: String,
            @field:SerializedName("course")
            @field:Expose
            var course: String,
            @field:SerializedName("degree")
            @field:Expose
            var degree: String,
            @field:SerializedName("from")
            @field:Expose
            var from: String,
            @field:SerializedName("to")
            @field:Expose
            var to: String) : Parcelable {

    @SerializedName("_id")
    @Expose
    var _id: ObjectId? = null

    constructor(parcel: Parcel) : this(
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString(),
            parcel.readString()) {
        _id = parcel.readParcelable(ObjectId::class.java.classLoader)
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(school)
        parcel.writeString(course)
        parcel.writeString(degree)
        parcel.writeString(from)
        parcel.writeString(to)
        parcel.writeParcelable(_id, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Education> {
        override fun createFromParcel(parcel: Parcel): Education {
            return Education(parcel)
        }

        override fun newArray(size: Int): Array<Education?> {
            return arrayOfNulls(size)
        }
    }
}
